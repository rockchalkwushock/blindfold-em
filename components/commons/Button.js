const Button = ({ condition, fn, text }) => (
  <button disabled={condition} onClick={() => fn()}>
    {text}
  </button>
)

export default Button
