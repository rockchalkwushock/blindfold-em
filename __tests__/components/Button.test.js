import React from 'react'
import { mount } from 'enzyme'

import Button from '../../components/commons/Button'

describe('Component: <Button />', () => {
  describe('Next:', () => {
    test('should mount without exploding in disabled state', () => {
      const tree = mount(<Button condition fn={() => {}} text="Next" />)
      expect(tree).toMatchSnapshot()
    })
    test('should mount without exploding in enabled state', () => {
      const tree = mount(<Button condition={false} fn={() => {}} text="Next" />)
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Prev:', () => {
    test('should mount without exploding  in disabled state', () => {
      const tree = mount(<Button condition fn={() => {}} text="Prev" />)
      expect(tree).toMatchSnapshot()
    })
    test('should mount without exploding  in enabled state', () => {
      const tree = mount(<Button condition={false} fn={() => {}} text="Prev" />)
      expect(tree).toMatchSnapshot()
    })
  })
})
