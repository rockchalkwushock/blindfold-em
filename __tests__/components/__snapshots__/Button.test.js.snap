// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Button /> Next: should mount without exploding in disabled state 1`] = `
<Button
  condition={true}
  fn={[Function]}
  text="Next"
>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Next
  </button>
</Button>
`;

exports[`Component: <Button /> Next: should mount without exploding in enabled state 1`] = `
<Button
  condition={false}
  fn={[Function]}
  text="Next"
>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Next
  </button>
</Button>
`;

exports[`Component: <Button /> Prev: should mount without exploding  in disabled state 1`] = `
<Button
  condition={true}
  fn={[Function]}
  text="Prev"
>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Prev
  </button>
</Button>
`;

exports[`Component: <Button /> Prev: should mount without exploding  in enabled state 1`] = `
<Button
  condition={false}
  fn={[Function]}
  text="Prev"
>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Prev
  </button>
</Button>
`;
